docs/tutorials/web.md

```diff
@@ -5,10 +5,421 @@ has_children: false
+
+```javascript
+function imageDataToTensor(image: Jimp, dims: number[]): Tensor {
+    // 1. Get buffer data from image and create R, G, and B arrays.
+    var imageBufferData = image.bitmap.data;
+    const [redArray, greenArray, blueArray] = new Array(new Array<number>(), Array<number>(), Array<number>());
+
+    // 2. Loop through the image buffer and extract the R, G, and B channels
+    for (let i = 0; i < imageBufferData.length; i += 4) {
+      redArray.push(imageBufferData[i]);
+      greenArray.push(imageBufferData[i + 1]);
+      blueArray.push(imageBufferData[i + 2]);
+      // skip data[i + 3] to filter out the alpha channel
+    }
+
+    // 3. Concatenate RGB to transpose [224, 224, 3] -> [3, 224, 224] to a number array
+    const transposedData = redArray.concat(greenArray).concat(blueArray);
+
+    // 4. convert to float32
+    let i, l = transposedData.length; // length, we need this for the loop
+    // create the Float32Array size 3 * 224 * 224 for these dimensions output
+    const float32Data = new Float32Array(dims[1] * dims[2] * dims[3]);
+    for (i = 0; i < l; i++) {
+      float32Data[i] = transposedData[i] / 255.0; // convert to float
+    }
+    // 5. create the tensor object from onnxruntime-web.
+      const inputTensor = new Tensor("float32", float32Data, dims);
+      return inputTensor;
+  }
+
+```
+### modelHelper.ts
+{: .no_toc }
+The inputTensor is ready for inferencing. Let's call the default `modelHelper.ts` function and walk through the logic. First we create the `ort.InferenceSession` by sending in the path to the model and the `SessionOptions`. For the `executionProviders` you can use either `webgl` to use the GPU or `wasm` to use the CPU. See the documentation to learn more about the `SessionOptions` available for inferencing configuration [here](https://onnxruntime.ai/docs/api/js/interfaces/InferenceSession.SessionOptions.html).
+
+```javascript
+import * as ort from 'onnxruntime-web';
+import _ from 'lodash';
+import {imagenetClasses} from '../data/imagenet';
+import { InferenceSession } from 'onnxruntime-web';
+
+const [predictions, inferenceTime] = await runSqueezenetModel(imageTensor);
+
+export async function runSqueezenetModel(preprocessedData: any): Promise<[any, number]> {
+    
+    // Create session and set options. See the docs here for more options: 
+    //https://onnxruntime.ai/docs/api/js/interfaces/InferenceSession.SessionOptions.html#graphOptimizationLevel
+
+    const session = await ort.InferenceSession
+                            .create('./_next/static/chunks/pages/squeezenet1_1.onnx', 
+                            { executionProviders: ['webgl'], graphOptimizationLevel: 'all' });
```

Comment:
`graphOptimizationLevel: 'all'` actually does not apply to webgl. suggest to remove

Reply:
we don't have an example for 'wasm' execution provider. do you want to add this ?

it will be helpful because you can demo users that they need to call:
```
ort.env.wasm.wasmPaths = {
   'ort-wasm.wasm': './_next/static/chunks/pages/ort-wasm.wasm',
   'ort-wasm-simd.wasm': './_next/static/chunks/pages/ort-wasm-simd.wasm'
};
```
before creating inference session

Reply:
This code works for both webgl and wasm, just update the config. I call this out in the text above the code " For the executionProviders you can use either webgl to use the GPU or wasm to use the CPU. See the documentation to learn more about the SessionOptions available for inferencing configuration here."

As for removing the `graphOptimizationLevel` I also think this should stay. Its not documented anywhere and I had to go into the source code to see how it works. Even if its not used here its good to have it documented.